core/coreapi/interface/interface.go

```diff
@@ -55,6 +58,37 @@ type UnixfsAPI interface {
 	Ls(context.Context, Path) ([]*Link, error)
 }
 
+// DagAPI specifies the interface to IPLD
+type DagAPI interface {
+	// Put inserts data using specified format and input encoding.
+	// If format is not specified (nil), default dag-cbor/sha256 is used
+	Put(ctx context.Context, src io.Reader, opts ...options.DagPutOption) ([]Node, error)
```

Comment:
Why is it desirable to have `Put()` return a slide of `Node`s?

Reply:
I guess it could return a slice of cids instead. But we need to know what was added.

Reply:
I mean why is it desirable that it's a slice -- isn't there a root node for the added dag?

Reply:
hrm... I think all of our current usecases would be fine with just the 'root' cid. Can you confirm that @magik6k ?

Reply:
It will be fine, yes. I decided to return a slice as it is what was returned by coredag.ParseInputs, I think the reasoning behind it was that the reader can provide more than one node, though it's not implemented now.

Reply:
Mh okay but what's the scenario that makes us want a slice returned here, versus just calling `Tree()` with the `Node.Cid()` returned from `Put()`? We'll keep all those nodes around in memory to return them from a function whose purpose isn't to iterate a graph (that's what `Tree()` is for).

I feel like we can rather be [consistent with js-ipfs](https://github.com/ipfs/interface-ipfs-core/blob/master/SPEC/DAG.md#dagput) here.